src/dex/cables/cables.ts

```diff
@@ -0,0 +1,1006 @@
+    let sumQuoteQty = 0;
+    const selectedRows: string[][] = [];
+
+    const isBase = qtyMode;
+    const isQuote = !qtyMode;
+
+    for (const [price, volume] of prices) {
+      if (isBase) {
+        if (sumBaseQty >= requiredQty) {
+          break;
+        }
+      }
+
+      if (isQuote) {
+        if (sumQuoteQty >= requiredQty) {
+          break;
+        }
+      }
+
+      let currentBaseQty = Number(volume);
+      let currentQuoteQty = Number(volume) * Number(price);
+
+      const overQty = isBase
+        ? currentBaseQty + sumBaseQty > requiredQty
+        : currentQuoteQty + sumQuoteQty > requiredQty;
+
+      if (overQty) {
+        if (isBase) {
+          currentBaseQty = requiredQty - sumBaseQty;
+          currentQuoteQty = currentBaseQty * Number(price);
+        }
+
+        if (isQuote) {
+          currentQuoteQty = requiredQty - sumQuoteQty;
+          currentBaseQty =
+            currentQuoteQty *
+            new BigNumber(1).dividedBy(new BigNumber(price)).toNumber();
+        }
+      }
+
+      sumBaseQty += currentBaseQty;
+      sumQuoteQty += currentQuoteQty;
+      selectedRows.push([price, currentBaseQty.toString()]);
+    }
+
+    const vSumBase = selectedRows.reduce((sum: number, [price, volume]) => {
+      return sum + Number(price) * Number(volume);
+    }, 0);
+
+    const price = new BigNumber(vSumBase)
+      .dividedBy(new BigNumber(sumBaseQty))
+      .toNumber();
+
+    if (isBase) {
+      return requiredQty / price;
+    } else {
+      return requiredQty * price;
+    }
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+    transferFees?: TransferFeeParams,
+    isFirstSwap?: boolean,
+  ): Promise<ExchangePrices<CablesData> | null> {
+    const isRestricted = await this.isRestricted();
+    if (isRestricted) {
+      return null;
+    }
+
+    await this.setTokensMap();
+
+    try {
+      const normalizedSrcToken = this.normalizeToken(srcToken);
+      const normalizedDestToken = this.normalizeToken(destToken);
+      // If: same token, return null
+      if (
+        normalizedSrcToken.address.toLowerCase() ===
+        normalizedDestToken.address.toLowerCase()
+      ) {
+        return null;
+      }
+
+      for (const symbol of Object.keys(this.tokensMap)) {
+        const normalizedTokenAddress =
+          this.tokensMap[symbol].address.toLowerCase();
+
+        if (normalizedSrcToken.address === normalizedTokenAddress) {
+          normalizedSrcToken.symbol = this.tokensMap[symbol].symbol;
+        }
+        if (normalizedDestToken.address === normalizedTokenAddress) {
+          normalizedDestToken.symbol = this.tokensMap[symbol].symbol;
+        }
+      }
+
+      // ---------- Pools ----------
+      let pools = await this.getPoolIdentifiers(
+        srcToken,
+        destToken,
+        side,
+        blockNumber,
+      );
```

Comment:
We can use passed `limitPools`, no need in making `getPoolIdentifiers` call again

Reply:
fixed